python/ray/ml/train/integrations/torch/torch_trainer.py

```diff
@@ -1,53 +1,169 @@
+            train.get_dataset_shard("my_dataset")
+
+            # Returns the total number of workers executing training.
+            train.get_world_size()
+
+            # Returns the rank of this worker.
+            train.get_world_rank()
+
+            # Returns the rank of the worker on the current node.
+            train.get_local_rank()
+
+    You can also use any of the :ref:`Torch specific function utils
+    <train-api-torch-utils>`.
+
+    .. code-block:: python
+        def train_loop_per_worker():
+            # Prepares model for distribted training by wrapping in
+            # `DistributedDataParallel` and moving to correct device.
+            train.torch.prepare_model(...)
+
+            # Configures the dataloader for distributed training by adding a
+            # `DistributedSampler`.
+            # You should NOT use this if you are doing
+            # `train.get_dataset_shard(...).to_torch(...)`
+            train.torch.prepare_data_loader(...)
+
+            # Returns the current torch device.
+            train.torch.get_device()
+
+    To save a model to use for the ``TorchPredictor``, you must save it under the
+    "model" kwarg in ``train.save_checkpoint()``.
+
+    Example:
+        .. code-block:: python
+
+            import torch
+            import torch.nn as nn
+
+            import ray
+            from ray import train
+            from ray.ml.integrations.torch import TorchTrainer
+
+            input_size = 1
+            layer_size = 15
+            output_size = 1
+
+            class NeuralNetwork(nn.Module):
+                def __init__(self):
+                    super(NeuralNetwork, self).__init__()
+                    self.layer1 = nn.Linear(input_size, layer_size)
+                    self.relu = nn.ReLU()
+                    self.layer2 = nn.Linear(layer_size, output_size)
+
+                def forward(self, input):
+                    return self.layer2(self.relu(self.layer1(input)))
+
+            def train_loop_for_worker():
+                dataset_shard = train.get_dataset_shard("train")
+                model = NeuralNetwork()
+                loss_fn = nn.MSELoss()
+                optimizer = optim.SGD(model.parameters(), lr=0.1)
+
+                model = train.torch.prepare_model(model)
+
+                for epoch in range(num_epochs):
+                    for batch in iter(dataset_shard.to_torch(batch_size=32)):
+                        output = model(input)
+                        loss = loss_fn(output, labels)
+                        optimizer.zero_grad()
+                        loss.backward()
+                        optimizer.step()
+                        print(f"epoch: {epoch}, loss: {loss.item()}")
+
+                    train.save_checkpoint(model=model.state_dict())
+
+            train_dataset = ray.data.from_items([1, 2, 3])
+            trainer = TorchTrainer(scaling_config={"num_workers": 3},
+                datasets={"train": train_dataset})
```

Comment:
shouldn't this be `train_dataset`?

Reply:
No, we ended up deciding tp change this back to just a single datasets arg yesterday.